lib/popover.js

```diff
@@ -0,0 +1,311 @@
+  `,
+  topRight: css`
+    right: 0;
+    bottom: 100%;
+  `,
+};
+
+const PopoverInner = styled.div`
+  position: relative;
+  overflow: hidden;
+  background-color: var(--colors-background);
+  font-size: var(--fontSizes-small);
+  border-radius: var(--rounded);
+
+  border: 1px solid var(--colors-border);
+  box-shadow: var(--popShadow);
+  z-index: 1;
+`;
+
+const PopoverContent = ({ align, children, ...props }) => {
+  const { ref, offset } = usePositionAdjustment();
+  return (
+    <PopoverWrap data-module="Popover" align={align} {...props}>
+      <PopoverInner ref={ref} style={offset}>
+        {children}
+      </PopoverInner>
+    </PopoverWrap>
+  );
+};
+
+const PopoverContainer = styled.details`
+  position: relative;
+  display: inline-block;
+`;
+const PopoverToggle = styled.summary`
+  list-style-image: none;
+  &::-webkit-details-marker {
+    display: none;
+    background: none;
+    color: transparent;
+    width: 0;
+  }
+`;
+
+const useClickOutside = (open, onClickOutside) => {
+  const ref = React.useRef();
+  const onClickOutsideProxy = useCallbackProxy(onClickOutside);
+  React.useEffect(() => {
+    if (!open) return;
+    const handler = (e) => {
+      if (document.body.contains(e.target) && !ref.current.contains(e.target)) {
+        onClickOutsideProxy(e);
+      }
+    };
+    window.addEventListener('click', handler);
+    return () => {
+      window.removeEventListener('click', handler);
+    };
+  }, [open, onClickOutsideProxy]);
+  return ref;
+};
+
+const useStack = (defaultState) => {
+  const [stack, setStack] = React.useState([defaultState]);
+  const top = stack[stack.length - 1];
+  const push = (value) => setStack((prev) => prev.concat([value]));
+  const pop = () => setStack((prev) => prev.slice(0, -1));
+  const clear = () => setStack([defaultState]);
+  return { top, push, pop, clear };
+};
+
+export const Popover = ({ align, label, views = {}, initialView, children, contentProps, ...props }) => {
+  const [open, setOpen] = React.useState(false);
+  const focusedOnMount = React.useRef();
+  const { top: activeView, push: setActiveView, pop: onBack, clear } = useStack(initialView);
+  const activeViewFunc = views[activeView] || children;
+  const onOpen = () => {
+    setOpen(true);
+    clear();
+  };
+  const onClose = () => {
+    setOpen(false);
+    clear();
+  };
+  React.useEffect(() => {
+    if (open && focusedOnMount.current) focusedOnMount.current.focus();
+  }, [open, activeView]);
+
+  const rootRef = useClickOutside(open, onClose);
+
+  const onToggle = (e) => {
+    // NOTE: this runs _after_ it toggles; we're updating the React state so that it matches the internal state
+    if (e.target.open) {
+      onOpen();
+    } else {
+      onClose();
+    }
+  };
+
+  useEscape(onClose);
+
+  // see https://www.w3.org/TR/wai-aria-practices-1.1/#menubutton
+  return (
+    <PopoverContainer data-module="PopoverMenu" ref={rootRef} onToggle={onToggle} open={open} role="menu" {...props}>
```

Comment:
Using `role="menu'` here implies also needing to set some roles on the children that are missing. 
The menu should also have aria-label. 
https://www.w3.org/TR/wai-aria-practices-1.1/#menu has details

It's not clear to me though that our popover containers are generally menus - the sign-in pop is for example but the delete project popover isn't really.

Reply:
should a popover always have an aria-label, or _only_ if it has role="menu"?

Reply:
Only if it has role='menu'